sample/redo/lib/main.dart

```diff
@@ -183,50 +183,32 @@ class _MyHomePageState extends State<MyHomePage> {
     // await _init();
   }
 
-  Mutator<void, Map<String, dynamic>> _todoCreateMutator;
-  Mutator<void, Map<String, dynamic>> _todoDoneMutator;
-  Mutator<void, Map<String, dynamic>> _todoRemoveMutator;
-  Mutator<void, Map<String, dynamic>> _todoReorderMutator;
+  Mutator<void, Map<String, dynamic>> _createTodo;
+  Mutator<void, Map<String, dynamic>> _removeTodo;
+  Mutator<void, Map<String, dynamic>> _updateTodo;
 
   _registerMutations() {
-    _todoCreateMutator = _replicache.register('todo-create',
+    _createTodo = _replicache.register('createTodo',
         (tx, Map<String, dynamic> args) async {
-      int id = args['id'];
-      String text = args['text'];
-      int listId = args['listId'];
-      Iterable<Todo> todos = await todosInListFromTx(tx, listId);
-      final order = newOrderBetween(todos.isEmpty ? null : todos.last, null);
-      await _write(tx, Todo(id, listId, text, false, order));
+      await _write(tx, Todo.fromJson(args));
     });
 
-    _todoDoneMutator = _replicache.register('todo-done',
-        (tx, Map<String, dynamic> args) async {
-      int id = args['id'];
-      bool complete = args['complete'];
-      final todo = await _read(tx, id);
-      if (todo == null) {
-        return;
-      }
-      todo.complete = complete;
-      await _write(tx, todo);
-    });
-
-    _todoRemoveMutator = _replicache.register('todo-remove', (tx, args) async {
+    _removeTodo = _replicache.register('removeTodo', (tx, args) async {
       int id = args['id'];
       await _del(tx, id);
     });
 
-    _todoReorderMutator =
-        _replicache.register('todo-reorder', (tx, args) async {
+    _updateTodo = _replicache.register('updateTodo', (tx, args) async {
       int id = args['id'];
-      double order = args['order'];
       final todo = await _read(tx, id);
       if (todo == null) {
         print('Warning: Possible conflict - Specified Todo $id is not present.'
             ' Skipping reorder.');
         return;
       }
-      todo.order = order;
+      todo.text ??= args['text'] ?? todo.text;
+      todo.complete = args['complete'] ?? todo.complete;
+      todo.order = args['order'] ?? todo.order;
```

Comment:
[taste] This is correct, but I think the read/write pair in the context of replay is a bit difficult to reason about. Consider adjusting `write` to only write the supplied fields.

Reply:
On second thought I take this back. Transactions definitely make this approach correct, and it's simple.

Reply:
`_write` takes a `Todo`, Not a "diff". If I did what you are suggesting I would have to read the `Todo` again inside `_write`. 

The `write` you seem to be suggesting seems to be what the `updateTodo_` function is doing?

Reply:
You're write (sorry not sorry).

Reply:
(I was thinking that the individual keys of a todo were stored separately in the kv store, not as a single value. nvm.)